### ISSUE TITLE
Possibility to extend rivets expression syntax

### ISSUE THREAD

**Issue by stalniy** [mikeric/rivets#591]
Currently only keypath expressions and primitives can be passed as parameters to both components' attributes and binders value. I suggest to allow users to extend this functionality. So, I for example want to pass an object to binder:
```js
<div rv-popover="enabled: user.isActive, template: customTemplate">{ user.name }</div>

**Comment by stalniy** [comment]
CC: @blikblum @Duder-onomy @Leeds-eBooks @mikeric @jccazeaux

**Comment by stalniy** [comment]
Please pay attention that I do not propose to add custom expression parsers in rivets-core but instead I propose to create extensibility point. So, people who wants to use more complex expression can do that

**Comment by jccazeaux** [comment]
Will a new (or extension of) parser be enough? If the parser decides it's an object, the binding mecanism will have have to know how to handle an object. Currently binding can only handle primitives and keypath types.
Is it really possible with only an extension point ?

**Comment by stalniy** [comment]
@jccazeaux 

yes, it's possible. So, look at the definition `rv-popover="enabled: user.isActive, template: myTemplate"`. We want to subscribe to values of the object declaration and compute a new object each time any of its values are changed. And values are basically keypaths or primitives which rivets are able to parse

**Comment by jccazeaux** [comment]
@stalniy I read again your proposal, I didn't understand it well.
I like your example, handling objects could be usefull and not so hard to do. 
Is it really necessary to open an extension point ?
